File: examples/dict_assign_typed.py
from typing import Dict

def test_string_dict() -> Dict[str, str]:
    """Test dictionary assignment with string keys"""
    d: Dict[str, str] = {}
    d["key1"] = "value1"
    d["key2"] = "value2"
    return d

def test_int_dict() -> Dict[int, str]:
    """Test dictionary assignment with integer keys"""
    d: Dict[int, str] = {}
    d[42] = "number key"
    d[100] = "another number"
    return d

def test_nested_dict() -> Dict[str, Dict[str, str]]:
    """Test nested dictionary (but not nested assignment yet)"""
    d: Dict[str, Dict[str, str]] = {}
    d["outer"] = {}
    # d["outer"]["inner"] = "value"  # TODO: This requires chained assignment
    return d

File: examples/dict_assign_typed.rs
use std::collections::HashMap;
#[doc = "Test dictionary assignment with string keys"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_string_dict() -> HashMap<String, String> {
    let mut d: HashMap<String, String> = {
        let map = HashMap::new();
        map
    };
    d.insert("key1".to_string().to_string(), "value1".to_string());
    d.insert("key2".to_string().to_string(), "value2".to_string());
    d
}
#[doc = "Test dictionary assignment with integer keys"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_int_dict() -> HashMap<i32, String> {
    let mut d: HashMap<i32, String> = {
        let map = HashMap::new();
        map
    };
    d.insert(42, "number key".to_string());
    d.insert(100, "another number".to_string());
    d
}
#[doc = "Test nested dictionary(but not nested assignment yet)"]
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_nested_dict() -> HashMap<String, HashMap<String, String>> {
    let mut d: HashMap<String, HashMap<String, String>> = {
        let map = HashMap::new();
        map
    };
    d.insert("outer".to_string().to_string(), {
        let map = HashMap::new();
        map
    });
    d
}
